Source/Parallelization/WarpXComm.cpp

```diff
@@ -90,19 +129,34 @@ WarpX::UpdateAuxilaryDataStagToNodal ()
         Array4<Real const> const& ey_fp = Efield_fp[0][1]->const_array(mfi);
         Array4<Real const> const& ez_fp = Efield_fp[0][2]->const_array(mfi);
 
-        const Box& bx = mfi.fabbox();
-        amrex::ParallelFor(bx,
-        [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
+        Box bx = mfi.validbox();
+        // TODO It seems like it is necessary to loop over the valid box grown
+        // with 2 guard cells. Should this number of guard cells be expressed
+        // in terms of the parameters defined in the guardCellManager class?
+        bx.grow(2);
+        amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
         {
+// FDTD variant
+#ifndef WARPX_USE_PSATD
             warpx_interp_nd_bfield_x(j,k,l, bx_aux, bx_fp);
             warpx_interp_nd_bfield_y(j,k,l, by_aux, by_fp);
             warpx_interp_nd_bfield_z(j,k,l, bz_aux, bz_fp);
             warpx_interp_nd_efield_x(j,k,l, ex_aux, ex_fp);
             warpx_interp_nd_efield_y(j,k,l, ey_aux, ey_fp);
             warpx_interp_nd_efield_z(j,k,l, ez_aux, ez_fp);
+// PSATD variant
+#else
+            warpx_interp_nd_bfield_x(j,k,l, bx_aux, bx_fp, fg_noy, fg_noz, p_stencil_coef_y, p_stencil_coef_z);
+            warpx_interp_nd_bfield_y(j,k,l, by_aux, by_fp, fg_nox, fg_noz, p_stencil_coef_x, p_stencil_coef_z);
+            warpx_interp_nd_bfield_z(j,k,l, bz_aux, bz_fp, fg_nox, fg_noy, p_stencil_coef_x, p_stencil_coef_y);
+            warpx_interp_nd_efield_x(j,k,l, ex_aux, ex_fp, fg_nox, p_stencil_coef_x);
+            warpx_interp_nd_efield_y(j,k,l, ey_aux, ey_fp, fg_noy, p_stencil_coef_y);
+            warpx_interp_nd_efield_z(j,k,l, ez_aux, ez_fp, fg_noz, p_stencil_coef_z);
+#endif
         });
     }
 
+    // NOTE: high-order interpolation is not implemented for mesh refinement
```

Comment:
Could we add an `amrex::Assert` to check that `field_gathering_nox == 2` (same in `y` and `z`) when `finest_level>0`?